include/bitcoin/blockchain/interface/block_chain.hpp

```diff
@@ -52,6 +53,7 @@ class BCB_API block_chain
     block_chain(threadpool& pool,
         const blockchain::settings& chain_settings,
         const database::settings& database_settings,
+        std::mutex& gbt_mutex,
```

Comment:
this mutex have to be a value member (not a reference nor a pointer) of block_chain